.github/workflows/buildimages-update.yml

```diff
@@ -19,43 +19,54 @@ on:
         description: 'Whether the images are test images'
         required: true
         type: boolean
+      include_otel_modules:
+        description: 'Whether to also bump the Go version in modules used by OpenTelemetry'
+        required: true
+        type: boolean
+
+permissions: {}
 
 jobs:
   open-go-update-pr:
     runs-on: ubuntu-latest
     permissions:
       contents: write # push commit and branch
+      pull-requests: write
 
     steps:
       - name: Checkout branch
-        uses: actions/checkout@v4
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
         with:
-          ref: ${{ github.head_ref }}
+          # credentials are needed to create the PR at the end of the workflow
+          persist-credentials: true
 
       - name: Fetch branch
+        env:
+          TARGET_BRANCH: ${{ inputs.branch }}
         # this step needs the github repository to be already cloned locally
         id: branch_fetch
         run: |
-          if git fetch origin "refs/heads/${{ inputs.branch }}"; then
+          if git fetch origin "refs/heads/$TARGET_BRANCH"; then
             echo "RESULT=true" >> $GITHUB_OUTPUT
           else
             echo "RESULT=false" >> $GITHUB_OUTPUT
           fi
 
       - name: Checkout branch
-        uses: actions/checkout@v4
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
```

Comment:
Could we use `actions/checkout@v4.1.4` instead ?

Reply:
Using the hash is a security good practise (and dependabot manages this correctly, by updating the comment when the hash changes)